### PR TITLE
iOS: getToken()

### PR DESCRIPTION
**Pull request by mikepitre**
> [!IMPORTANT]
> 🔎 Previews:
>
> - https://clerk.com/docs/pr/1387/references/ios/get-token

### Explanation:

- ❓ Looking for some advice on how to organize this. Should this be considered a guide or something else? It's more of a callout of an important SDK function, and how and why to use it.
### This PR:

- Adds a section to the iOS SDK reference for getting a user's session token for use with a customer's backend.

### CODE REVIEW

**Comment by alexisintech** [comment]
This looks like a reference, and it looks like you organized it correctly. Are there other iOS SDK methods that need to be documented like this?

File: docs/references/ios/get-token.mdx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+func getToken(
+  _ options: GetTokenOptions = .init()
+) async throws -> TokenResource?
+```
+
+## Parameters
+
+| Name | Type | Description |
+| - | - | - |
+| `options` | `GetTokenOptions` | Options that can be passed as parameters to the `getToken()` function. |
```
Comment (alexisintech) [inline docs/references/ios/get-token.mdx:24]: I think this is the correct syntax

```suggestion
<Properties>
  - `options`
  - [`GetTokenOptions`](#get-token-options)
 
  Options that can be passed as parameters to the `getToken()` function.
</Properties>
```

File: docs/references/ios/get-token.mdx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+### `GetTokenOptions`
+
+| Name | Type | Description |
+| - | - | - |
+| `template` | `String?` | The name of the JWT template from the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=jwt-templates) to generate a new token from.<br />For example: 'firebase', 'grafbase', or your custom template's name. |
+| `expirationBuffer` | `Double` | If the cached token will expire within `{expirationBuffer}` seconds, fetch a new token instead. Max is 60 seconds. Defaults to 10 seconds. |
+| `skipCache` | `Bool` | Whether to skip the cache lookup and force a refresh of the token instead. Useful if the token claims are time-sensitive or depend on data that can be updated (e.g. user fields). Defaults to false. |
```
Comment (alexisintech) [inline docs/references/ios/get-token.mdx:32]: use the [`<Properties>` component](https://github.com/clerk/clerk-docs/blob/main/CONTRIBUTING.md#properties-1) here as well

**Comment by mikepitre** [comment]
> This looks like a reference, and it looks like you organized it correctly. Are there other iOS SDK methods that need to be documented like this?

Nope, not yet anyway! Should this go under a category or is it fine right on the same level as the overview?

File: docs/references/ios/get-token.mdx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+---
+title: getToken()
+description: Use Clerk's iOS SDK to retrieve a token for a JWT Template that is defined in the Clerk Dashboard.
+---
+
+# `getToken()`
+
+Retrieves the user's session token for the default Clerk token or the given template.
```
Comment (alexisintech) [inline docs/references/ios/get-token.mdx:8]:
```suggestion
Retrieves the user's session token for the [default Clerk token](/docs/backend-requests/resources/session-tokens) or the given [template](/docs/backend-requests/making/jwt-templates).
```

**Review by alexisintech — APPROVED** [review]

**Pull request merged**